### DISCUSSION TITLE
Redpanda Kafka container starts anyway with `quarkus.kafka-streams.bootstrap-servers` configured

### DISCUSSION THREAD

**Discussion by ppatierno** [quarkusio/quarkus#22539] [Q&A]
From the doc I read that one of the ways you can avoid starting the Redpanda container during Kafka development is by:

```
the kafka.bootstrap.servers is configured
```

That is true but why it's not the same with the `quarkus.kafka-streams.bootstrap-servers` when developing with Kafka Streams?

[a comment by an automated account is not shown]

**Reply by cescoffier** [reply]
The dev service is handled by the kafka-client extension, and not by the kafka-streams extension.
Note that the kafka-streams extension also understands `kafka.bootstrap.servers`.

**Reply by ppatierno** [reply]
> The dev service is handled by the kafka-client extension

That's something that the Quarkus devs know, not the users.
Users want just write an application using Kafka Streams and don't need to explicitly using kafka-client extension.

> Note that the kafka-streams extension also understands kafka.bootstrap.servers

Yeah, I know that but then what's the point of having `quarkus.kafka-streams.bootstrap-servers` as well, as possible config param?

**Reply by cescoffier** [reply]
The kafka-streams extension depends on the kafkac-lient extension, so users do not have to know.

About `quarkus.kafka-streams.bootstrap-servers`, this was the original property that we still support. However, I would recommend using `kafka.bootstrap.servers` as it covers all Kafka extensions and also supports the service binding operators. I agree that documentation should be updated.
